runtime/ops/mod.rs

```diff
@@ -37,18 +38,24 @@ use std::cell::RefCell;
 use std::future::Future;
 use std::rc::Rc;
 
-pub fn reg_json_async<F, R>(rt: &mut JsRuntime, name: &'static str, op_fn: F)
-where
-  F: Fn(Rc<RefCell<OpState>>, Value, BufVec) -> R + 'static,
-  R: Future<Output = Result<Value, AnyError>> + 'static,
+pub fn reg_json_async<F, V, R, RV>(
+  rt: &mut JsRuntime,
+  name: &'static str,
+  op_fn: F,
+) where
+  F: Fn(Rc<RefCell<OpState>>, V, BufVec) -> R + 'static,
+  V: DeserializeOwned,
+  R: Future<Output = Result<RV, AnyError>> + 'static,
+  RV: Serialize,
```

Comment:
To be honest I find this signature way harder to comprehend...

Reply:
We are just aligning the signature to core here. This will let us get rid of a bunch of `serde_json::from_value` and `serde_json::to_value` in WebGPU op crate. Hopefully will let us reduce LOC count by a few hundred.